File: ios/VietBridge/Views/Cards/RiskCard.swift
import SwiftUI

struct RiskCard: View {
    let data: ChatResponseData

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            HStack(spacing: 4) {
                Image(systemName: "shield.fill")
                    .font(.system(size: 12))
                    .foregroundStyle(.red)
                Text("风险分析")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundStyle(.red)
            }

            if let score = data.score {
                HStack {
                    RiskGauge(score: score)
                        .frame(width: 80, height: 80)
                    VStack(alignment: .leading, spacing: 4) {
                        Text(riskLevel(score))
                            .font(.system(size: 16, weight: .bold))
                            .foregroundStyle(scoreColor(score))
                        Text("风险评分 \(score)/100")
                            .font(.system(size: 12))
                            .foregroundStyle(.textSecondary)
                    }
                    Spacer()
                }
            }

            if let situation = data.situation {
                Text(situation)
                    .font(.system(size: 13))
                    .foregroundStyle(.textPrimary)
                    .padding(10)
                    .background(Color.bgInput)
                    .clipShape(RoundedRectangle(cornerRadius: 8))
            }

            if let factors = data.factors, !factors.isEmpty {
                Text("风险因素")
                    .font(.system(size: 13, weight: .bold))
                    .foregroundStyle(.textPrimary)
                ForEach(factors) { f in
                    HStack(spacing: 8) {
                        Text(f.label)
                            .font(.system(size: 12))
                            .foregroundStyle(.textPrimary)
                            .frame(maxWidth: .infinity, alignment: .leading)
                        GeometryReader { geo in
                            RoundedRectangle(cornerRadius: 3)
                                .fill(scoreColor(f.weight))
                                .frame(width: geo.size.width * CGFloat(f.weight) / 100)
                        }
                        .frame(width: 60, height: 6)
                    }
                }
            }

            if let tips = data.tips, !tips.isEmpty {
                Text("防范建议")
                    .font(.system(size: 13, weight: .bold))
                    .foregroundStyle(.textPrimary)
                ForEach(tips, id: \.self) { tip in
                    Text("• \(tip)")
                        .font(.system(size: 12))
                        .foregroundStyle(.textSecondary)
                }
            }

            if let scripts = data.scripts, !scripts.isEmpty {
                Text("实用话术")
                    .font(.system(size: 13, weight: .bold))
                    .foregroundStyle(.textPrimary)
                ForEach(scripts) { s in
                    VStack(alignment: .leading, spacing: 4) {
                        Text(s.vi)
                            .font(.system(size: 14, weight: .medium))
                        Text(s.zh)
                            .font(.system(size: 12))
                            .foregroundStyle(.textSecondary)
                        ActionButtons(text: s.vi)
                    }
                    .padding(8)
                    .background(Color.bgInput)
                    .clipShape(RoundedRectangle(cornerRadius: 8))
                }
            }
        }
        .padding(16)
        .cardStyle()
    }

    private func riskLevel(_ score: Int) -> String {
        if score >= 70 { return "高风险" }
        if score >= 40 { return "中风险" }
        return "低风险"
    }

    private func scoreColor(_ score: Int) -> Color {
        if score >= 70 { return .riskRed }
        if score >= 40 { return .riskOrange }
        return .vbAccent
    }
}

struct RiskGauge: View {
    let score: Int

    var body: some View {
        ZStack {
            Circle()
                .stroke(Color.borderLight, lineWidth: 6)
            Circle()
                .trim(from: 0, to: CGFloat(score) / 100)
                .stroke(gaugeColor, style: StrokeStyle(lineWidth: 6, lineCap: .round))
                .rotationEffect(.degrees(-90))
            Text("\(score)")
                .font(.system(size: 20, weight: .bold))
                .foregroundStyle(gaugeColor)
        }
    }

    private var gaugeColor: Color {
        if score >= 70 { return .riskRed }
        if score >= 40 { return .riskOrange }
        return .vbAccent
    }
}

File: android/app/src/main/java/com/vietbridge/ai/ui/cards/RiskCard.kt
package com.vietbridge.ai.ui.cards

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Lightbulb
import androidx.compose.material.icons.filled.Shield
import androidx.compose.material.icons.filled.VolumeUp
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.vietbridge.ai.data.model.ChatResponseData
import com.vietbridge.ai.ui.components.ActionButton
import com.vietbridge.ai.ui.components.VBCard
import com.vietbridge.ai.ui.theme.*
import com.vietbridge.ai.util.TTSHelper

@Composable
fun RiskCard(data: ChatResponseData) {
    VBCard {
        Column(Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(12.dp)) {
            Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                Icon(Icons.Default.Shield, null, Modifier.size(14.dp), tint = RiskAccentColor)
                Text("风险评估", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = RiskAccentColor)
            }

            // Gauge
            data.score?.let { score ->
                Box(Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
                    RiskGauge(score, data.level ?: "unknown")
                }
            }

            // Factors
            data.factors?.takeIf { it.isNotEmpty() }?.let { factors ->
                Text("风险因素", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = TextSecondary)
                factors.forEach { factor ->
                    Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                        Box(Modifier.padding(top = 6.dp).size(6.dp).clip(CircleShape).background(RiskOrange))
                        Column {
                            Text(factor.label, fontWeight = FontWeight.Bold, fontSize = 14.sp)
                            factor.detail?.let { Text(it, fontSize = 12.sp, color = TextSecondary) }
                        }
                    }
                }
            }

            // Tips
            data.tips?.takeIf { it.isNotEmpty() }?.let { tips ->
                Text("防护建议", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = TextSecondary)
                tips.forEach { tip ->
                    Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                        Icon(Icons.Default.Lightbulb, null, Modifier.size(14.dp), tint = RiskYellow)
                        Text(tip, fontSize = 12.sp, color = TextPrimary)
                    }
                }
            }

            // Scripts
            data.scripts?.takeIf { it.isNotEmpty() }?.let { scripts ->
                Text("应对话术", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = TextSecondary)
                scripts.forEach { script ->
                    Column(Modifier.fillMaxWidth().clip(RoundedCornerShape(8.dp)).background(BgInput).padding(8.dp)) {
                        script.situation?.let { Text(it, fontSize = 11.sp, color = TextTertiary) }
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Column(Modifier.weight(1f)) {
                                Text(script.vi, fontWeight = FontWeight.Bold, fontSize = 14.sp)
                                Text(script.zh, fontSize = 12.sp, color = TextSecondary)
                            }
                            ActionButton(Icons.Default.VolumeUp) { TTSHelper.speakVietnamese(script.vi) }
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun RiskGauge(score: Int, level: String) {
    val color = when {
        score < 30 -> VBAccent
        score < 60 -> RiskYellow
        score < 80 -> RiskOrange
        else -> RiskRed
    }
    val levelLabel = when (level.lowercase()) {
        "safe" -> "安全"; "low" -> "低风险"; "medium" -> "中风险"; "high" -> "高风险"; "critical" -> "极高风险"; else -> "未知"
    }

    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Box(Modifier.size(100.dp), contentAlignment = Alignment.Center) {
            Canvas(Modifier.fillMaxSize()) {
                val stroke = Stroke(width = 10.dp.toPx(), cap = StrokeCap.Round)
                drawArc(BorderLight, 0f, 360f, false, style = stroke, topLeft = Offset.Zero, size = Size(size.width, size.height))
                drawArc(color, -90f, 360f * score / 100f, false, style = stroke, topLeft = Offset.Zero, size = Size(size.width, size.height))
            }
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text("$score", fontSize = 24.sp, fontWeight = FontWeight.Bold, color = color)
                Text("/ 100", fontSize = 10.sp, color = TextTertiary)
            }
        }
        Spacer(Modifier.height(4.dp))
        Text(levelLabel, fontSize = 13.sp, fontWeight = FontWeight.Bold, color = color)
    }
}
